crates/sui-graphql-rpc/src/context_data/db_query_cost.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use diesel::{
+    pg::Pg,
+    query_builder::{bind_collector::RawBytesBindCollector, QueryBuilder, QueryFragment},
+    PgConnection, RunQueryDsl,
+};
+use regex::Regex;
+use sui_indexer::schema_v2::query_cost;
+
+/// Extracts the raw sql query string from a diesel query
+/// and replaces all the parameters with '0'
+pub fn raw_sql_string_values_set(
+    query: &dyn QueryFragment<Pg>,
+) -> Result<String, crate::error::Error> {
+    // This lets us get the underlying sql query param types & values
+    // Currently unused because we set all vals to '0' but want to keep for future reference
+
+    // let mut bind_collector = RawBytesBindCollector::<Pg>::new();
+    // QueryFragment::<Pg>::collect_binds(
+    //     &query,
+    //     &mut bind_collector,
+    //     pg_connection as &mut dyn diesel::pg::PgMetadataLookup,
+    //     &Pg,
+    // )?;
+
+    // // These hold the real values of the parameters
+    // // but we cannot repr them as needed since we dont have proper
+    // // type info
+    // // So use '0' for all
+    // // Todo: could using shorter strings imply lower cost e.g for searching? We dont want that
+    // let _binds = &bind_collector.binds;
+    // let metadata = &bind_collector.metadata;
+    // let num_params = metadata.len();
+
+    let mut query_builder = <diesel::pg::Pg as diesel::backend::Backend>::QueryBuilder::default();
+    QueryFragment::<Pg>::to_sql(&query, &mut query_builder, &Pg).map_err(|e| {
+        crate::error::Error::Internal(format!(
+            "Failed to extract raw sql string from query: {}",
+            e
+        ))
+    })?;
+    let sql: String = query_builder.finish();
+
+    let re = Regex::new(r"\$(\d+)")
+        .map_err(|e| crate::error::Error::Internal(format!("Failed create valid regex: {}", e)))?;
+    Ok(re.replace_all(&sql, "'0'").to_string())
```

Comment:
Tried it. Didnt work. The bindings are returned as vec<u8> for all types which isnt always castable because we dont know the positions each type goes into. '0' seemed to work for all types I tried though